modules/local/subworkflow/quantify_salmon.nf

```diff
@@ -0,0 +1,95 @@
+params.genome_options       = [:]
+params.salmon_index_options = [:]
+params.salmon_quant_options = [:]
+params.merge_counts_options = [:]
+
+include { GUNZIP              } from '../process/gunzip'                        addParams( options: params.genome_options       )
+include { UNTAR               } from '../process/untar'                         addParams( options: params.index_options        )
+include { GTF_GENE_FILTER     } from '../process/gtf_gene_filter'               addParams( options: params.genome_options       )
+include { GFFREAD             } from '../process/gffread'                       addParams( options: params.genome_options       )
+include { SALMON_INDEX        } from '../../nf-core/software/salmon/index/main' addParams( options: params.salmon_index_options )
+include { SALMON_QUANT        } from '../../nf-core/software/salmon/quant/main' addParams( options: params.salmon_quant_options )
+include { SALMON_TX2GENE      } from '../process/salmon_tx2gene'                addParams( options: params.genome_options       )
+include { SALMON_TXIMPORT     } from '../process/salmon_tximport'               addParams( options: [publish_by_id : true]      )
+include { SALMON_MERGE_COUNTS } from '../process/salmon_merge_counts'           addParams( options: params.merge_counts_options )
+include { SALMON_SUMMARIZEDEXPERIMENT as SALMON_SE_GENE 
+          SALMON_SUMMARIZEDEXPERIMENT as SALMON_SE_TRANSCRIPT } from '../process/salmon_summarizedexperiment' addParams( options: params.merge_counts_options )
+
+workflow QUANTIFY_SALMON {
+    take:
+    reads            // channel: [ val(meta), [ reads ] ]
+    index            //    file: /path/to/salmon/index/
+    transcript_fasta //    file: /path/to/transcript.fasta
+    genome_fasta     //    file: /path/to/genome.fasta
+    gtf              //    file: /path/to/genome.gtf
+
+    main:
+    /*
+     * Uncompress Salmon index or generate from scratch if required
+     */
+    if (index) {
+        if (index.endsWith('.tar.gz')) {
+            ch_index = UNTAR ( index ).untar
+        } else {
+            ch_index = file(index)
+        }
+    } else {
+        if (transcript_fasta) {
+            if (transcript_fasta.endsWith('.gz')) {
+                ch_transcript_fasta = GUNZIP ( transcript_fasta ).gunzip
+            } else {
+                ch_transcript_fasta = file(transcript_fasta)
+            }
+        } else {
+            ch_transcript_fasta = GFFREAD ( genome_fasta, GTF_GENE_FILTER ( genome_fasta, gtf ) ).fasta
+        }
+        ch_index = SALMON_INDEX ( ch_transcript_fasta ).index
+    }
+
+    /*
+     * Quantify and merge counts across samples
+     */
+    SALMON_QUANT        ( reads, ch_index, gtf )
+    SALMON_TX2GENE      ( SALMON_QUANT.out.results.collect{it[1]}, gtf )
+    SALMON_TXIMPORT     ( SALMON_QUANT.out.results, SALMON_TX2GENE.out.collect() )
+    SALMON_MERGE_COUNTS (
+        SALMON_TXIMPORT.out.counts_gene.collect{it[1]},

```

Comment:
The `[1]` might need some details on why we need this ;-)

Reply:
[This](https://github.com/drpatelh/rnaseq/commit/2772403a1139c8681d1c3ec9b2325116ef12f48c) ok?